### PR TITLE
Fixing progress for Get-ComputerInfo

### PR DESCRIPTION
**Pull request by powercode**
# PR Summary

The function that handled progress for Get-ComputerInfo always set the RecordType to Completed.

## PR Context

Progress for the cmdlet seemed to disappear.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by PoshChan-Staging** [comment]
@powercode, your last commit had 10 failures in `PowerShell-CI-windows`
### Tests for Get-ComputerInfo: Ensure Type returned.Verify type returned by Get-ComputerInfo
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1: line 1010
1010:             $computerInfo = Get-ComputerInfo

```
### Tests for Get-ComputerInfo.Validate All Properties.Error occurred in Context block
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
At D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1:32 char:39
```
### Tests for Get-ComputerInfo.Filter Variations.Test 02.001 Filter Property - Property filter with one valid item
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1: line 25
25:         return Get-ComputerInfo -Property $properties

```
### Tests for Get-ComputerInfo.Filter Variations.Test 02.002 Filter Property - Property filter with three valid items
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1: line 25
25:         return Get-ComputerInfo -Property $properties

```
### Tests for Get-ComputerInfo.Filter Variations.Test 02.005 Filter Property - Property filter with valid and invalid items: ver #1
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1: line 25
25:         return Get-ComputerInfo -Property $properties

```
### Tests for Get-ComputerInfo.Filter Variations.Test 02.006 Filter Property - Property filter with valid and invalid items: ver #2
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1: line 25
25:         return Get-ComputerInfo -Property $properties

```
### Tests for Get-ComputerInfo.Filter Variations.Test 02.007 Filter Property - Property filter with wild card: ver #1
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1: line 25
25:         return Get-ComputerInfo -Property $properties

```
### Tests for Get-ComputerInfo.Filter Variations.Test 02.008 Filter Property - Property filter with wild card and fixed
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1: line 25
25:         return Get-ComputerInfo -Property $properties

```
### Tests for Get-ComputerInfo.Filter Variations.Test 02.009 Filter Property - Property filter with wild card, fixed and invalid
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1: line 25
25:         return Get-ComputerInfo -Property $properties

```
### Special Case Tests for Get-ComputerInfo.Error occurred in Describe block
  Error: Cannot process the argument because the value of statusDescription cannot be null or empty.
```stacktrace
At D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Get-ComputerInfo.Tests.ps1:32 char:39
```

File: src/Microsoft.PowerShell.Commands.Management/commands/management/GetComputerInfoCommand.cs [hunk 1]
```diff
@@ -224,18 +223,14 @@ protected override void ProcessRecord()
...
+        {       
+            status = status ?? ComputerResources.ProgressStatusCompleted;
+            var progress = new ProgressRecord(0, activity, status);                                
+            if (status == null)
             {
-                _progress = new ProgressRecord(0, activity, status);
-                WriteProgress(_progress);
+                progress.RecordType = ProgressRecordType.Completed;
             }
+            WriteProgress(progress);
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/GetComputerInfoCommand.cs]:
```suggestion

            WriteProgress(progress);
```

**Review by SteveL-MSFT — APPROVED** [review]
Minor style nit

File: src/Microsoft.PowerShell.Commands.Management/commands/management/GetComputerInfoCommand.cs (hunk 1 above)
Comment (powercode) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/GetComputerInfoCommand.cs]: fixing

**Comment by SteveL-MSFT** [comment]
@powercode I believe you're the first person to get to experience @PoshChan's new capability.  I'm curious of your thoughts if it was helpful and if you have any suggestions to improve it.

**Comment by powercode** [comment]
@SteveL-MSFT  I loved the easy access to the test failures! Didn't think much about what should be improved as I just saw it as a huge improvement!

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@powercode I was wondering whether a test like the following should be added:

https://github.com/PowerShell/PowerShell/blob/665cd26ab0f29bf8703b0cb922a54b8fe6c89c1e/test/powershell/engine/Help/HelpSystem.Tests.ps1#L99-110

**Comment by powercode** [comment]
@adityapatwardhan  You're welcome to push it!

**Comment by adityapatwardhan** [comment]
@powercode Found a bug while adding test. The value of `status` was updated before the `null` check, hence we never set the `recordtype` as `completed`. Fixed the bug and added the test. Please have a look.

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**